### PR TITLE
Improve Krylov subspace solver in CPHF

### PR DESCRIPTION
**Pull request by wxj6000**
- Use standard block krylov subspace solver instead of 1d krylov subspace solver
- Fixed the issue in sg1_prune

### CODE REVIEW

**Comment by wxj6000** [comment]
@sunqm Krylov subspace solver is improved in this PR. And most CPHF converge in 4-6 steps.

**Comment by sunqm** [comment]
> @sunqm Krylov subspace solver is improved in this PR. And most CPHF converge in 4-6 steps.

Why can CPHF converge faster? It's not clear for me from the changes in this PR.

**Comment by wxj6000** [comment]
> > @sunqm Krylov subspace solver is improved in this PR. And most CPHF converge in 4-6 steps.
> 
> Why can CPHF converge faster? It's not clear for me from the changes in this PR.

@sunqm I added some other changes too. Previously, mo1base is flattened. So the number of variables for the linear system was 3*natm*nao*nocc. It slows down CPSCF because 1) more number of variables than needed. 2) it does not take the advantage of collaboration of multiple RHS.

File: gpu4pyscf/df/df.py [hunk 1]
```diff
@@ -30,7 +30,9 @@
 
 MIN_BLK_SIZE = getattr(__config__, 'min_ao_blksize', 128)
 ALIGNED = getattr(__config__, 'ao_aligned', 32)
-LINEAR_DEP_TOL = incore.LINEAR_DEP_THR
+
+# TODO: reuse the setting in pyscf 2.6
+LINEAR_DEP_THR = 1e-9#incore.LINEAR_DEP_THR
```
Comment (wxj6000) [inline gpu4pyscf/df/df.py:35]: change it to 1e-6

File: gpu4pyscf/dft/gen_grid.py [hunk 2]
```diff
@@ -72,12 +72,13 @@ def sg1_prune(nuc, rads, n_ang, radii=radi.SG1RADII):
...
 #            6  38 86  194 86
-    leb_ngrid = numpy.array([6, 38, 86, 194, 86])
-    alphas = numpy.array((
+    leb_ngrid = cupy.array([6, 38, 86, 194, 86])
+    alphas = cupy.array((
         (0.25  , 0.5, 1.0, 4.5),
         (0.1667, 0.5, 0.9, 3.5),
         (0.1   , 0.4, 0.8, 2.5)))
     r_atom = radii[nuc] + 1e-200
+    rads = cupy.asarray(rads)
```
Comment (wxj6000) [inline gpu4pyscf/dft/gen_grid.py:81]: Add unit test for sg1 grids

File: gpu4pyscf/scf/cphf.py [hunk 3]
```diff
@@ -70,7 +70,8 @@ def vind_vo(mo1):
 
 # h1 shape is (:,nocc+nvir,nocc)
 def solve_withs1(fvind, mo_energy, mo_occ, h1, s1,
-                 max_cycle=50, tol=1e-9, hermi=False, verbose=logger.WARN):
+                max_cycle=50, tol=1e-9, hermi=False,
+                verbose=logger.WARN, level_shift=0):
```
Comment (wxj6000) [inline gpu4pyscf/scf/cphf.py:74]: enable level_shift in hessian class

File: gpu4pyscf/dft/gen_grid.py (hunk 2 above)
Comment (wxj6000) [reply gpu4pyscf/dft/gen_grid.py:81]: Will be added in https://github.com/pyscf/gpu4pyscf/pull/158

**Pull request merged**